MSBot/src/models/azureBotService.ts

```diff
@@ -9,10 +9,19 @@ export class AzureBotService extends ConnectedService implements IAzureBotServic
 
     constructor(source: Partial<IAzureBotService> = {}) {
         super(source);
-        const {  tenantId='', subscriptionId='', resourceGroup='' } = source;
-        this.tenantId = tenantId;
-        this.subscriptionId = subscriptionId;
-        this.resourceGroup = resourceGroup;
+        const { tenantId = '', subscriptionId = '', resourceGroup = '' } = source;
+        Object.assign(this, { tenantId, subscriptionId, resourceGroup });
+
+        let { id } = this;
+        Object.defineProperty(this, 'id', {
+            get: function () {
+                return id || tenantId;
```

Comment:
>tenantId [](start = 29, length = 8)

this is not right.  the id is the unique value, or we should add a botId  and then make id <- botId